File: app/src/main/java/com/kota/Bahamut/Command/BahamutCommandGoodArticle.java
package com.kota.Bahamut.Command;

import androidx.annotation.NonNull;

import com.kota.Bahamut.ListPage.TelnetListPage;
import com.kota.Bahamut.ListPage.TelnetListPageBlock;
import com.kota.Telnet.TelnetClient;

public class BahamutCommandGoodArticle extends TelnetCommand {
    int _article_index;

    public BahamutCommandGoodArticle(int articleIndex) {
        this._article_index = articleIndex;
        this.Action = GoodArticle;
    }

    public void execute(TelnetListPage aListPage) {
        if (this._article_index > 0) {
            TelnetClient.getClient().sendStringToServer(this._article_index + "\ngy");
        }
    }

    public void executeFinished(TelnetListPage aListPage, TelnetListPageBlock aPageData) {
        setDone(true);
    }

    @NonNull
    public String toString() {
        return "[GoodArticle][articleIndex=" + this._article_index + "]";
    }
}

File: app/src/main/java/com/kota/Bahamut/Command/BahamutCommandGoodArticle.kt
package com.kota.Bahamut.Command

import com.kota.Bahamut.ListPage.TelnetListPage
import com.kota.Bahamut.ListPage.TelnetListPageBlock
import com.kota.Telnet.TelnetClient

class BahamutCommandGoodArticle(var _article_index: Int) : TelnetCommand() {
    init {
        this.Action = BahamutCommandDefs.Companion.GoodArticle
    }

    override fun execute(aListPage: TelnetListPage?) {
        if (this._article_index > 0) {
            TelnetClient.getClient().sendStringToServer(this._article_index.toString() + "\ngy")
        }
    }

    override fun executeFinished(aListPage: TelnetListPage?, aPageData: TelnetListPageBlock?) {
        setDone(true)
    }

    override fun toString(): String {
        return "[GoodArticle][articleIndex=" + this._article_index + "]"
    }
}
